core/extension/gdextension_interface.h

```diff
@@ -2109,6 +2109,33 @@ typedef GDExtensionVariantPtr (*GDExtensionInterfaceDictionaryOperatorIndex)(GDE
  */
 typedef GDExtensionVariantPtr (*GDExtensionInterfaceDictionaryOperatorIndexConst)(GDExtensionConstTypePtr p_self, GDExtensionConstVariantPtr p_key);
 
+/**
+ * @name dictionary_ref
+ * @since 4.2
+ *
+ * Sets a Dictionary to be a reference to another Dictionary object.
+ *
+ * @param p_self A pointer to the Dictionary object to update.
+ * @param p_from A pointer to the Dictionary object to reference.
+ */
+typedef void (*GDExtensionInterfaceDictionaryRef)(GDExtensionTypePtr p_self, GDExtensionConstTypePtr p_from);
```

Comment:
Could you elaborate (in comment) when this function is supposed to be used by GDExtension? Especially in contrast to the copy constructor.

(I understand it exists for arrays as well, but might also benefit from some docs there).

Reply:
Also needs an update of `@since`, as this feature won't make it to 4.2.

Reply:
…You know, the more I look into it, the less I'm certain. It was initially a matter of achieving parity with the typed arrays so I just didn't question it, but I don't think this explicit setup is all that necessary. I'll probably hammer out a PR to get a feel for if it's needed

Reply:
Removed under the current assumption that `Dictionary::Operator=` will be able to replace this (see [this comment](https://github.com/godotengine/godot/pull/82497#issuecomment-1742150126))